### PR TITLE
Bugfix storage pointer addressspace. 

### PR DESCRIPTION
**Pull request by veriAjinkya**
Fixed following bug: 
Self struct itself was declared in storage address space but pointers in the self struct were pointing to default addressspace.

### CODE REVIEW

**Review by Technius — COMMENTED** [review]
Needs rebasing, and a few tests need to be updated.

File: lib/include/libyul2llvm/YulASTVisitor/IntrinsicHelper.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ class YulIntrinsicHelper {
   llvm::Value *getPointerToStorageVarByName(std::string,
                                             llvm::Instruction *insertPoint);
   llvm::StringRef getStorageVarYulTypeByName(llvm::StringRef name);
-  llvm::Type *getTypeByTypeName(llvm::StringRef type);
+  llvm::Type *getTypeByTypeName(llvm::StringRef type, int addrSpaceId=0);
```
Comment (Technius) [inline lib/include/libyul2llvm/YulASTVisitor/IntrinsicHelper.h:37]: nit: should use a constant variable and not 0 here.

**Review by Technius — APPROVED** [review]
LGTM besides a nitpick

File: lib/include/libyul2llvm/YulASTVisitor/IntrinsicHelper.h (hunk 1 above)
Comment (veriAjinkya) [reply lib/include/libyul2llvm/YulASTVisitor/IntrinsicHelper.h]: done

**Pull request merged**